querybook/webapp/components/QueryEditor/QueryEditor.tsx

```diff
@@ -368,11 +368,33 @@ export class QueryEditor extends React.PureComponent<
         });
     }
 
+    @bind
+    public async prefetchDataTables(tableReferences: TableToken[]) {
+        const { getTableByName } = this.props;
+        if (!getTableByName) {
+            return;
+        }
+
+        const tablesGettingLoaded = new Set();
+
+        const tableLoadPromises = [];
+        for (const { schema, name } of tableReferences) {
+            const fullName = `${schema}.${name}`;
+            if (!tablesGettingLoaded.has(fullName)) {
+                tableLoadPromises.push(getTableByName(schema, name));
+                tablesGettingLoaded.add(fullName);
+            }
+        }
+
+        return Promise.all(tableLoadPromises);
```

Comment:
return is not needed?

Reply:
return is needed. otherwise, the `getContextSensitiveWarnings` wont wait for the promises to be resolved

Reply:
it would be needed if you don't have await, actually, in this case, you don't need the async since await is never used